eventhubs/client/azure-eventhubs/src/test/java/com/azure/eventhubs/EventPositionTest.java

```diff
@@ -0,0 +1,254 @@
+        Assert.assertEquals(offsetEvent.sequenceNumber(), enqueuedTimeEvent.sequenceNumber());
+    }
+
+    /**
+     * Test for receiving offset without inclusive filter
+     */
+    @Ignore
+    @Test
+    public void testReceiverOffsetNonInclusiveFilter() {
+        skipIfNotRecordMode();
+
+        // Arrange
+        final EventPosition enqueuedTimeEventPosition = EventPosition.fromEnqueuedTime(Instant.EPOCH);
+        final EventData enqueuedTimeEvent = receiveMessageHelper(enqueuedTimeEventPosition, 1).toIterable().iterator().next();
+
+        final EventPosition offsetEventPosition = EventPosition.fromOffset(enqueuedTimeEvent.offset(), false);
+        final EventData offsetEvent = receiveMessageHelper(offsetEventPosition, 1).toIterable().iterator().next();
+
+        // Assert
+        Assert.assertEquals(offsetEvent.sequenceNumber(), enqueuedTimeEvent.sequenceNumber() + 1);
+    }
+
+    /**
+     * Test for receiving sequence number with inclusive filter
+     */
+    @Ignore
+    @Test
+    public void testReceiverSequenceNumberInclusiveFilter() {
+        skipIfNotRecordMode();
+
+        // Arrange
+        final EventPosition enqueuedTimeEventPosition = EventPosition.fromEnqueuedTime(Instant.EPOCH);
+        final EventData enqueuedTimeEvent = receiveMessageHelper(enqueuedTimeEventPosition, 1).toIterable().iterator().next();
+
+        final EventPosition nextSequenceNumberEventPosition = EventPosition.fromSequenceNumber(enqueuedTimeEvent.sequenceNumber(), true);
+        final EventData sequenceNumEvent = receiveMessageHelper(nextSequenceNumberEventPosition, 1).toIterable().iterator().next();
+
+        // Assert
+        Assert.assertEquals(sequenceNumEvent.offset(), enqueuedTimeEvent.offset());
+        Assert.assertEquals(sequenceNumEvent.sequenceNumber(), enqueuedTimeEvent.sequenceNumber());
+    }
+
+    /**
+     * Test for receiving sequence number without inclusive filter
+     */
+    @Ignore
+    @Test
+    public void sequenceNumberNonInclusiveFilterFromEnqueuedTime() {
+        skipIfNotRecordMode();
+
+        // Arrange
+        final EventPosition enqueuedTimeEventPosition = EventPosition.fromEnqueuedTime(Instant.EPOCH);
+        final EventData enqueuedTimeEvent = receiveMessageHelper(enqueuedTimeEventPosition, 1).toIterable().iterator().next();
+
+        final EventPosition nextSequenceNumberEventPosition = EventPosition.fromSequenceNumber(enqueuedTimeEvent.sequenceNumber(), false);
+        final EventData sequenceNumberEvent = receiveMessageHelper(nextSequenceNumberEventPosition, 1).toIterable().iterator().next();
+
+        // Assert
+        Assert.assertEquals(sequenceNumberEvent.sequenceNumber(), enqueuedTimeEvent.sequenceNumber() + 1);
+    }
+
+
+    private Flux<EventData> receiveMessageHelper(EventPosition eventPosition, int numberOfEvents) {
+        receiver = client.createConsumer(getConsumerGroupName(), PARTITION_ID, eventPosition,
```

Comment:
Every time you call this method, you are setting a new `receiver`, but never disposing of the existing one that lives when we did `beforeTest`. Is there a reason for this? This method is doing two things again. Its better to pass in an existing consumer.